### ISSUE TITLE
Many issues

### ISSUE THREAD

**Issue by Harunoha** [elishacloud/Silent-Hill-2-Enhancements#26]
I was having trouble making this mod to work earlier, and was about to write about the mod not working for me, the white shader fix and 2d fog fix were nowhere to be seen, and maybe something else, but then I did SOMETHING and the white shader fix finally worked! yay! but... the 2d fog was still not there, and the noise filter on the screen disappeared too (now it just looks like a sepiaish filter is on screen when I activate it) , and that's not all, the performance of the game slowed down a lot at the same time, and yes, I have a pretty old PC, but the game was running smooth as butter before, and now the frames per second went down to like 11 in some parts, maybe less.

It's like I fixed 1 of 5 problems and now I have 3 new problems that didn't happen before

So, to summarize, I had some problems, 2d fog not present, white shader fix not working, the menu doesn't fill the entire screen either, BUT the game ran smooth 30fps
NOW I somehow made the white shader fix to work, just not the other ones, but now I can't see the noise filter (which I used so the screen is not that empty when there's no 2d fog), AND my game performance dropped significantly (which seems to be caused by the complex fog, since when I disable the complex fog the performance gets a little better, but still, as I said, It didn't happen before) all of this at the same time the white shader fix worked.

**Comment by Polymega** [comment]
Can you upload a screenshot here of your game's directory? Be sure to set the view as "Detailed" and the "Date last modified" table is also shown.

**Comment by Harunoha** [comment]
![untitled](https://user-images.githubusercontent.com/42750330/44698316-72cc5e80-aa4d-11e8-81f2-dacac79e86bc.png)

The files that are not in the frame are the icon and logo images, and the sh2pc.exe (2/28/2003 8:44 PM) and wined3d.dll (10/2/2015 10:52 AM)

**Comment by AeroWidescreen** [comment]
@Polymega Interestingly, he appears to be using Windows 8. It could be completely unrelated of course, but it's worth keeping note of.

**Comment by Polymega** [comment]
@Harunoha if you're using WineD3D it worries me how many `d3d9` files you have. I can't see all of their extensions but I'm guessing you have `d3d9.dll`, `d3d9.ini`, and `d3d9.log`? If that's correct, that means you're using the ReShade d3d9 files and NOT the WineD3D d3d9 files. 

What you'll need to do then is:

- Delete `d3d9.dll`, `d3d9.ini`, and `d3d9.log` (these files would be from ReShade DirectX)
- [Download WineD3D For Windows (version 1.7.52)](http://downloads.fdossena.com/Projects/WineD3D/Builds/WineD3DForWindows_1.7.52.zip)
- Copy `d3d9.dll`, `libwine.dll`, and `wined3d.dll` from the archive to `\KONAMI\Silent Hill 2\`
- If you wish to also use ReShade download[ ReShade (OpenGL) & SH2 PC PS2 Filter](http://enhanced.townofsilenthill.com/SH2/files/SH2PC_ReShade_3.4.0_OpenGL.zip)
- Copy the content of the `\Silent Hill 2\` folder from the archive to `\KONAMI\Silent Hill 2\`
   - This copies over `opengl32.dll`, `opengl32.ini`, and `ReShade Filter - SH2 PC PS2.ini`

I look forward to the day we can drop ReShade and not have to split the installation directions for users!

@AeroWidescreen definitely a bit of a trend going on here. Also, I hate to say this, but I'd almost prefer we work on fixing what WineD3D fixes without using it first over the fullscreen enhancements. _A lot_ of user tickets coming in relate to this fickle thing.

**Comment by Polymega** [comment]
Also, @Harunoha , on your computer go to Start > type "Folder Options" > Open the Folder Options window > click the "View" tab > and scroll down and uncheck "Hide extensions for known file types"

**Comment by Harunoha** [comment]
Ok, sorry for the late reply, if there are ReShade files in the folder is probably because I tried using it in the start, but then I saw the horrible performance and freaked out and deleted all ReShade files I could find.

I'm now going to re-install everything again from the start and report any changes

**Comment by Polymega** [comment]
Sure thing. And note: If you are **not** using an Nvidia GPU you don't need to use the WineD3D files (as that software is a bit of a performance hog). This is especially true if you're installing this on a non-gaming laptop: WineD3D doesn't normally play well with laptop hardware specs.

**Comment by Harunoha** [comment]
I completely re-installed it, and no luck, it's still the same, white shader fix working properly, but no 2d fog, no noise filter, and horrible performance (around 15 to 25 fps on the first street, and 10 or so when in the foggy forest trail)

**Comment by Harunoha** [comment]
And yeah, I have a crappy Nvidia card, and followed the instructions step by step

**Comment by Polymega** [comment]
> and the noise filter on the screen disappeared too (now it just looks like a sepiaish filter is on screen when I activate it)

Is there any way you can screenshot this particular issue? 

In regards to performance: The Fog Fix in particular will be one of the most demanding things required of your PC. I know this is not ideal, but for older PCs/aging graphics cards you could: 

- Disable the Fog Fix
- Ensure `IncreaseNoiseEffectRes =` is set no higher than `512`
- Set `d3d8to9 = 0` (this will also disable WineD3D)
- Ensure `FPSLimit =` is set to `30` and no higher
- Reduce the game's resolution via `ResX =` and `ResY =`

We're hoping to remove the need for WineD3D while still fixing in the game what WineD3D fixes. If we can do this, it will not only remove WineD3D from the equation, but also keep the game running as DirectX for Nvidia users, which should also help with compatibility even more.

**Comment by Harunoha** [comment]
ok I'll try what you stated above and yeah, I actually tried to take a screencap of it earlier but somehow couldn't, but it's really not much too, as I said, the screen goes a bit brown sepia and darker, looks like  when the noise filter is on but without the noise

**Comment by Harunoha** [comment]
Ok, I fiddled with the resolution a little bit and set it lower without disabling WineD3D and it improved somewhat the performance, it was not 30 fps yet, but at least more bearable, and somehow this made the noise filter work again too. I was about to just give up and play it like that, but it seems like the voice acting and sounds during in-game cutscenes are out of sync with the non 30 fps things going on and it was horrible still, super out of sync, so I gave up and disabled the WineD3D as you suggested and it improved the performance to the point where the in-game cutscenes are ok again too, but yeah, no white shader fix this time, guess I'll just play it like that then, hope it doesn't take out the experience that much

**Comment by Polymega** [comment]
> Ok, I fiddled with the resolution a little bit and [...] somehow this made the noise filter work again too.

Interesting. This is good to know for future troubleshooting. And, to be sure, you didn't also set the `IncreaseNoiseEffectRes =` resolution too high as well before, correct?

> but yeah, no white shader fix this time, guess I'll just play it like that then, hope it doesn't take out the experience that much

There are only three instances of the white shaders bug: The tunnel at the beginning on the game (at the Observation Deck), Woodside Apt. Room 209 & Room 101.

**Comment by AeroWidescreen** [comment]
@Harunoha 

> and somehow this made the noise filter work again too. 

Could you also tell us what _display_ resolution you were using before? Example: 1920x1080. It sounds like that could be a problem with IncreaseNoiseEffectRes but we need to verify first.

> I gave up and disabled the WineD3D as you suggested and it improved the performance to the point where the in-game cutscenes are ok again too, but yeah, no white shader fix this time, guess I'll just play it like that then, hope it doesn't take out the experience that much

I'm sorry to hear. Obviously your GPU plays a role here, but I think performance would be much better if the game wasn't limited to one core. Maybe in the future someone will solve that issue.

**Comment by Harunoha** [comment]
IncreaseNoiseEffectRes was set to 512 always, never touched it.

> There are only three instances of the white shaders bug: The tunnel at the beginning on the game (at the Observation Deck), Woodside Apt. Room 209 & Room 101.

Thanks for this, I thought it happened more often, like, all doors on rooms or something

**Comment by Polymega** [comment]
@AeroWidescreen I received an interesting comment on the demonstration trailer video: 
https://www.youtube.com/watch?v=6v1USVhtfjQ&lc=UgxEN0hKjZJce9wBgKZ4AaABAg

> I must say that this has been an improvement to the disgusting port they brought to the PC version, there is no other word to say, but many thanks for "remastering" the game, although I would like to see a future update on the FPS playable at 60 and the issue of the shadows ... now I don't need to establish the affinity of the cores with this graphical correction, thanks to the adjustment of the 3D hardware accelerator, very well operated in Windows 10 ... Good job!

Allegedly, by IndrectSound forcing the game's audio to run in Hardware 3D Mode it fixes the sound loop issue? Would you want to give this a test run on Windows 10 and I'll do the same for Windows 7 soon?

**Comment by SirChronocide** [comment]
Just tried installing everything as normal and I too am still having problems with the PS2 Noise Filter working. I've tried all recommendations so far in this thread (changing resolution, replacing DLL files, disabling WinD3D) and nothing seems to work; it just refuses to enable the noise filter.
